### PR TITLE
Animal Genetics

### PR DESCRIPTION
**Pull request by Azrun**
## About the PR 
Allows for geneticists to scan and collect genes from non-humanoid critters. The gene scanner has a new research item specifically called non-humanoid scanner. 

To facilitate the collection and scanning of these new Gene sources, pet carriers can now direct animals into the genetic scanner.

The assignment of genetic traits has been somewhat generalized in the determination of how many traits to gain to allow for future expansion of how these genes are determined. For the scope of this PR, genes are specifically a signed to various critters. 

The following types of critters now have some form of bio effect either already unlocked or scrambled inside them: mice, birds, cockroaches, scorpions, snakes, frogs, possums armadillos, fireflies, and bunnies.

As a rapper to make genetics abilities that reference existing targetable abilities. Hopefully this allows for minimal duplication of code while still utilizing the benefits of and limitations of geneticsAbilities and existing targetable abilities.

## Why's this needed? 
Give geneticists something to do? Expands the ability for away missions or terrainify to have some form of engagement with geneticists.

## Changelog 

```changelog
(u)Azrun
(*)Expands Genetics to scan some small animals! Stick 'em in a carrier and see what is out there.
```

### CODE REVIEW

**Comment by Glamurio** [comment]
This is an awesome idea. Anything for geneticists to leave their office more.

**Review by Glamurio — CHANGES_REQUESTED** [review]
A couple of typos etc.

File: code/modules/medical/genetics/bioEffects/beneficial.dm [hunk 1]
```diff
@@ -796,6 +796,150 @@ var/list/radio_brains = list()
...
+
+	OnRemove()
+		..()
+		if (ishuman(owner))
+			var/mob/living/carbon/human/H = owner
+			REMOVE_MOVEMENT_MODIFIER(H, /datum/movement_modifier/slither, src.type)
+
+/datum/bioEffect/food_stores
+	name = "Lipid Stores"
+	desc = "Subject gains the ability to improve the nourhsment available from their lipid stores."
```
Comment (Vaskritaya) [inline code/modules/medical/genetics/bioEffects/beneficial.dm:903]:
```suggestion
	desc = "Subject gains the ability to improve the nourishment available from their lipid stores."
```

File: code/modules/medical/genetics/bioEffects/beneficial.dm [hunk 2]
```diff
@@ -796,6 +796,150 @@ var/list/radio_brains = list()
...
+/obj/item/parts/human_parts/arm/right/claw/critter/pincer
+	limb_type = /datum/limb/small_critter/pincers
+
+/datum/bioEffect/carapace
+	name = "Chitinoarmis Durescutis "
+	desc = "Subject skin develops into a hardened carapace."
+	id = "carapace"
+	probability = 20
+	effectType = EFFECT_TYPE_POWER
+	msgGain = "You feel like skin becoming hardened."
```
Comment (Vaskritaya) [inline code/modules/medical/genetics/bioEffects/beneficial.dm:855]:
```suggestion
	msgGain = "You feel your skin harden."
```

File: code/modules/medical/genetics/bioEffects/powers.dm [hunk 3]
```diff
@@ -2302,6 +2302,99 @@
...
+	wrapped_ability = /datum/targetable/critter/wasp_sting/snake_bite
+	start_on_cooldown = TRUE
+	has_misfire = FALSE
+	needs_hands = FALSE
+	override_params = list("amt1"=5)
+
+
+/datum/bioEffect/power/critter/scorpion_sting
+	name = "Scorpiocauda Vipernox"
+	desc = "Generates a hardened keratin area between the mouth and nose."
```
Comment (Vaskritaya) [inline code/modules/medical/genetics/bioEffects/powers.dm:2346]: You forgot to add a description while copy-pasting this from beak.

**Review by Vaskritaya — COMMENTED** [review]
Some typos, etc.

File: code/modules/medical/genetics/bioEffects/powers.dm (hunk 3 above)
Comment (Azrun) [reply code/modules/medical/genetics/bioEffects/powers.dm]: 🤦 I made the PR, noticed this and thought I addressed it.... just didn't save the file. 😢

**Comment by Xkallubar** [comment]
You know what would be cool? A mutation that you can find on critters that, when activated, polls ghosts if they want to play as that creature. A sentience mutation. Would be fun, having a spider you found in space as a sentient pet.

**Comment by pali6** [comment]
I like the idea of animal genetics!

**Pull request merged**